httpd/filters.go

```diff
@@ -0,0 +1,147 @@
+package httpd
```

Comment:
I hear ya -- I've had the same struggles myself with source files becoming 5,6, 7K lines long. It's nuts. It's just that to me the functionality in this file peers deep into the handler object structure, doing stuff like `h.server.Authenticate()`, such that this file is far from a modular piece of code that can be easily unit tested (for example). To me, all this new file get us is 100-odd lines out of another file. It doesn't ease testing or make the code significantly easier to follow. I might be missing something though.